models/User/AbstractUser.php

```diff
@@ -45,7 +45,13 @@ public static function getById(int $id): static|null
             if (\Pimcore\Cache\RuntimeCache::isRegistered($cacheKey)) {
                 $user = \Pimcore\Cache\RuntimeCache::get($cacheKey);
             } else {
-                $user = new static();
+                $reflectionClass = new \ReflectionClass(static::class);
```

Comment:
Do we really need reflection here?
It seems like more of a problem in the UserController in the delete function, where we call the AbstractUser::getById directly. Shouldn't we tackle the problem there?
https://github.com/pimcore/pimcore/blob/11.x/bundles/AdminBundle/src/Controller/Admin/UserController.php#L246

Reply:
I think it should still be possible to load role or user with a parent class (both classes are static now). The same we did in Document to load via PageSnippet abstract class:
https://github.com/pimcore/pimcore/blob/b23b847c42a79fce4ed6905570283a04ae2b221f/models/Document.php#L196-L201